### ISSUE TITLE
Should a share group for initializer inputs of a derived dataset add up to a 100?

### ISSUE THREAD

**Issue by grdw** [quintel/etsource#1217]
Based on [this discussion](https://github.com/quintel/atlas/pull/86/files#r97404052)

---

I'm wondering if a share group for initializer inputs should add up to a 100. Should that be validated for a derived dataset. Example:

```bash
ls etsource/initializer_inputs/shares/

share_1.ad
share_2.ad
share_3.ad
share_4.ad
share_5.ad
```

Do share 1 till 5 *all* need to be applied in a derived dataset? If you want to be explicit about all the sliders of the slider group, than yes. But must you be?

Including @AlexanderWirtz @ChaelKruip @ploh

**Comment by ChaelKruip** [comment]
> But must you be?

What slider settings should you get for the sliders that are not explicitly set? If you leave that to the balancer, the result will be completely arbitrary and a scenario may not make any sense. I think that slider settings should define the scenario unambiguously.

**Comment by ChaelKruip** [comment]
Including @antw

**Comment by grdw** [comment]
> What slider settings should you get for the sliders that are not explicitly set?

Well like this example:

![example](https://camo.githubusercontent.com/b4220aa8a5453ae9adfb8badaeaa80595413ff1b/687474703a2f2f692e696d6775722e636f6d2f35346b3138474b2e706e67)

The coal fired heater was explicitly set to 10% and the rest of the shares were not explicitly touched.

**Comment by ChaelKruip** [comment]
> The coal fired heater was explicitly set to 10% and the rest of the shares were not explicitly touched.

I understand. My question was meant rhetorically 😆 

In your example, the (non-specified) sliders are either set to their default value (which will change with each dataset update), or set to a value by the balancer. The balancer has a certain logic as to which slider to use for the balancing which is not transparent for the user. That is a bad thing IMO.

**Comment by grdw** [comment]
> I understand. My question was meant rhetorically

That makes the whole comment a lot eassier to read 😅 

> The balancer has a certain logic as to which slider to use for the balancing which is not transparent for the user. That is a bad thing IMO.

I'm not aware of that, I just assumed that it picks the first slider out of the share group and compensates with that.

**Comment by antw** [comment]
> I'm not aware of that, I just assumed that it picks the first slider out of the share group and compensates with that.

In the ETModel front-end this is mostly true: it choses the first slider which the visitor has not already moved. ETEngine also has [an automatic group balancer](https://github.com/quintel/etengine/blob/master/app/models/balancer.rb) (via [Osmosis](https://github.com/quintel/osmosis)) which moves all sliders in the group "fairly" (defined by their `max-min` and their current value). This is mostly for the benefit of other applications (ETFlex, ETMoses).

```ruby
Osmosis.balance([
  { min: 0, max: 100, value: 50 }
  { min: 0, max: 100, value: 0 }
], 100)

# => [75, 25]
```

> If you leave that to the balancer, the result will be completely arbitrary and a scenario may not make any sense. I think that slider settings should define the scenario unambiguously.

I agree. For datasets, I think it best to be explicit. If you rely on the autobalancer there are two issues:

* Initializer inputs will need to have a min/max value
* If the min/max changes at a later date, the balancer will have a different outcome: the dataset is no longer immutable.

**Comment by DorinevanderVlies** [comment]
> I agree. For datasets, I think it best to be explicit.

I agree as well. 

The way we handle this now in the Excel files used to make a local dataset is by an internal check to see if the total group adds up to exactly 1.

**Comment by ploh** [comment]
Closed by quintel/atlas#86
